src/components/Tooltip/index.jsx

```diff
@@ -0,0 +1,54 @@
+import React from 'react';
+import { shape, string } from 'prop-types';
+import MuiTooltip from '@material-ui/core/Tooltip';
+import Typography from '@material-ui/core/Typography';
+import InfoIcon from '@material-ui/icons/Info';
+
+function Tooltip({ keyword, classes }) {
+  /**
+   * Generate tooltip descriptions to provide further information
+   * regarding the given keywords.
+   * (only keywords defined as complex object types will need
+   *  tooltip descriptions)
+   */
+  const tooltipDescriptions = {
+    additionalItems: 'Additional items must match a sub-schema',
+    additionalProperties: 'Additional properties must match a sub-schema',
+    dependencies:
+      'The schema of the object may change based on the presence of certain special properties',
+    propertyNames: 'Names of properties must follow a specified convention',
+    patternProperties:
+      'Property names or values should match the specified pattern',
+  };
+  const createTooltipTitle = key =>
+    `${tooltipDescriptions[key]}. See the JSON-schema source for details.`;
+
+  return (
+    <MuiTooltip title={createTooltipTitle(keyword)} arrow>
+      <div className={classes.line}>
+        <Typography
+          component="div"
+          variant="subtitle2"
+          className={classes.tooltip}>
+          {`${keyword}: `}
+        </Typography>

```

Comment:
When we're showing the icon, what do you think about removing the column and adding some margin (e.g., 4px instead)?

_Example: removing the column and adding some margin on propertyNames_

<img width="192" alt="Screen Shot 2019-12-31 at 4 42 36 PM" src="https://user-images.githubusercontent.com/3766511/71634757-cd9b3a80-2bec-11ea-94e1-245addd99d81.png">

Reply:
Great idea! Removed the colon from Tooltip's text.
It makes the UI look more cleaner 👍 

![image](https://user-images.githubusercontent.com/29671309/71637908-0e9a5a00-2c94-11ea-83d7-50265b9ab342.png)